src/main/java/net/snowflake/client/core/SFJsonResultSet.java

```diff
@@ -250,4 +256,242 @@ public Date getDate(int columnIndex, TimeZone tz) throws SFException {
+          return new SfSqlArray(
+              columnSubType, getStream(nodeElements, floatConverter).toArray(Float[]::new));
+        case Types.DOUBLE:
+          JsonStringToTypeConverter doubleConverter =
+              value -> converters.getNumberConverter().getDouble(value, Types.DOUBLE);
+          return new SfSqlArray(
+              columnSubType, getStream(nodeElements, doubleConverter).toArray(Double[]::new));
+        case Types.DATE:
+          JsonStringToTypeConverter dateConverter = createDateConverter();
+          return new SfSqlArray(
+              columnSubType, getStream(nodeElements, dateConverter).toArray(Date[]::new));
+        case Types.TIME:
+          JsonStringToTypeConverter timeConverter = createTimeConverter();
+          return new SfSqlArray(
+              columnSubType, getStream(nodeElements, timeConverter).toArray(Time[]::new));
+        case Types.TIMESTAMP:
+          JsonStringToTypeConverter timestampConverter =
+              createTimestampConverter(columnSubType, columnType, scale);
+          return new SfSqlArray(
+              columnSubType, getStream(nodeElements, timestampConverter).toArray(Timestamp[]::new));
+        case Types.BOOLEAN:
+          JsonStringToTypeConverter booleanConverter =
+              value -> converters.getBooleanConverter().getBoolean(value, columnType);
+          return new SfSqlArray(
+              columnSubType, getStream(nodeElements, booleanConverter).toArray(Boolean[]::new));
+        case Types.STRUCT:
+          JsonStringToTypeConverter structConverter = createStructConverter();
+          return new SfSqlArray(
+              columnSubType, getStream(nodeElements, structConverter).toArray(Map[]::new));
+        case Types.ARRAY:
+          JsonStringToTypeConverter arrayConverter = createArrayConverter();
+          return new SfSqlArray(
+              columnSubType, getStream(nodeElements, arrayConverter).toArray(Map[][]::new));
+        default:
+          return null;
+      }
+    } catch (JsonProcessingException e) {
+      throw new SFException(e, ErrorCode.INVALID_STRUCT_DATA);
+    }
+  }
+
+  private static JsonStringToTypeConverter createArrayConverter() {
+    JsonStringToTypeConverter arrayConverter =
+        value -> {
+          try {
+            return OBJECT_MAPPER.readValue(value, HashMap[].class);
+          } catch (JsonProcessingException e) {
+            throw new SFException(e, ErrorCode.INVALID_STRUCT_DATA);
+          }
+        };
+    return arrayConverter;
+  }
+
+  private static JsonStringToTypeConverter createStructConverter() {
+    JsonStringToTypeConverter structConverter =
+        value -> {
+          try {
+            return OBJECT_MAPPER.readValue(value, Map.class);
+          } catch (JsonProcessingException e) {
+            throw new SFException(e, ErrorCode.INVALID_STRUCT_DATA);
+          }
+        };
+    return structConverter;
```

Comment:
Can we inline this variable? Same in other methods?

Reply:
Ok